lib/system/sets.nim

```diff
@@ -13,9 +13,11 @@
 proc cardSetImpl(s: ptr UncheckedArray[uint8], len: int): int {.inline.} =
   var i = 0
   result = 0
+  var num = uint64(0)
```

Comment:
`{.noinit.}` really and it can safely be moved into the loop (which prevents usages from leaking outside of the loop accidentally

Reply:
Make sense.

Reply:
Nit pick, use

```suggestion
  var num = 0'u64
```